www/src/utils/node/docs.js

```diff
@@ -1,8 +1,34 @@
 const _ = require(`lodash`)
 const path = require(`path`)
 const { slash } = require(`gatsby-core-utils`)
+const minimatch = require(`minimatch`)
+
 const { getPrevAndNext } = require(`../get-prev-and-next.js`)
 const { getMdxContentSlug } = require(`../get-mdx-content-slug`)
+const findApiCalls = require(`../find-api-calls`)
+
+const ignorePatterns = [
+  `**/commonjs/**`,
+  `**/node_modules/**`,
+  `**/__tests__/**`,
+  `**/dist/**`,
+  `**/__mocks__/**`,
+  `babel.config.js`,
+  `graphql.js`,
+  `**/flow-typed/**`,
+]
+
+function isCodeFile(node) {
+  return (
+    node.internal.type === `File` &&
+    node.sourceInstanceName === `packages` &&
+    [`js`].includes(node.extension) &&
```

Comment:
This currently doesn't traverse TypeScript files (and throws a parsing error if we add `.ts` here). I'm afraid this plugin will eventually break and be useless as we refactor the docs source to TypeScript?

Reply:
Agreed. Seems like we'd need to address that anyway, though, independent of this change. I'm also not sure how quickly this site is likely to start incorporating more TS. That feels like a @gillkyle question!

Reply:
It's less about the site incorporating TS and more about the refactors of `packages/gatsby` into TypeScript. This function crawls the Gatsby source and finds all instances where a function is used. But since it only crawls JS files, all the gatsby source that has been converted to TypeScript will not be picked up.

Reply:
No problem. In retrospect, my comment doesn't even make sense. I think I was trying to look at too many PRs at once. Thanks for rolling with me! The first part, though, about needing to address this anyway... If I'm not mistaken, we're _currently_ only crawling JS files, true?

Reply:
Yes, we are. But this is going to cause bugs (if it hasn't already).

So it's not necessarily in scope for this PR, but something we'll have to deal with.

Reply:
Not sure if this will fix all the issues, but `flow` and `typescript` babel plugins are incompatible (you can't use both of them at the same time). When I was writing this parser we didn't have typescript so I hard coded `flow` there:

https://github.com/gatsbyjs/gatsby/blob/342d835b6e02e766fd3a2abd9cb7c4fc7daf28cc/www/plugins/gatsby-transformer-gatsby-api-calls/find-api-calls.js#L27

In gatsby core we use very similar setup for parser (when we analyze files for page and static queries and extract them):

https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/utils/babel-parse-to-ast.ts 

You can check there that we replace `flow` plugin for `typescript` if file we are checking is `ts(x)` (see `getBabelParserOptions` function in file linked above) - we could do similar thing here. It might not make it work (not sure if AST checking code will work or not after that), but it should fix parsing errors